plotly/offline/offline.py

```diff
@@ -327,9 +330,9 @@ def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly',
     if not ipython:
         raise ImportError('`iplot` can only run inside an IPython Notebook.')
 
-    config = {}
-    config['showLink'] = show_link
-    config['linkText'] = link_text
+    config = dict(config) if config else {}
+    config.setdefault('showLink', show_link)
```

Comment:
The reason for [`dict.setdefault`](https://docs.python.org/2/library/stdtypes.html#dict.setdefault) is that it doesn't set the value if it is already set. This gives user the option to construct the _whole_ view configuration dict beforehand, not needing to worry about having `showLink` overwritten or needing to pass it separately as `show_link`. I'd rather go with:
```py
if 'linkText' not in config:
    config['linkText'] = link_text
```
but I think `.setdefault()` is pretty standard. Do you insist? The behavior of not overriding the option if it's already set is documented.